config/vufind/searches.ini

```diff
@@ -300,16 +310,22 @@ WorkKeys = year
 ;       who owns a subject guide whose title most closely matches the search terms.
 ;       Uses configuration in LibGuidesAPI.ini and requires configuration in
 ;       contentsecuritypolicy.ini.
-; LibGuidesResults:[GET parameter]:[result limit]
+; LibGuidesResults:[GET parameter]:[result limit]:[ini section]:[custom heading]
 ;       Display LibGuides research guides search results matching the terms found in
 ;       the specified GET parameter (default = "lookfor"), limited to a specified
 ;       number of matches (default = 5).
+;       Hidden filters may be defined in the specified section of the LibGuides.ini
+;       file; see [CatalogResultsVideoFilter] below for an example of this. The
+;       optional custom heading overrides the default for this recommendation module.
 ; LibGuidesResultsDeferred:[GET parameter]:[result limit]
 ;       Same as LibGuidesResults, but loaded via AJAX.
-; LibGuidesAZResults:[GET parameter]:[result limit]
+; LibGuidesAZResults:[GET parameter]:[result limit]:[ini section]:[custom heading]
 ;       Display LibGuides A-Z Databases search results matching the terms found in
 ;       the specified GET parameter (default = "lookfor"), limited to a specified
 ;       number of matches (default = 5).
+;       Hidden filters may be defined in the specified section of the LibGuidesAZ.ini
+;       file; see [CatalogResultsVideoFilter] below for an example of this. The
+;       optional custom heading overrides the default for this recommendation module.
```

Comment:
I've added the documentation to these other modules for discussion purposes.  Two obvious problems though.

1. [custom heading] doesn't work until I implement it in the templates.  I can either add it to the templates or remove it from the doc for now.
2. Filtering doesn't work for many of these backends -- definitely not for LibGuides / LibGuidesAZ, not yet implemented for EPF, and I don't know about Web.  I assume it's implemented for Summon but can't test that.  So I'll presumably remove the [ini section] from those but wanted to get your take first.

Reply:
1.) I think you might as well go ahead and implement the custom headings for all of the templates. I wonder if it would actually make sense to just create a $heading property and getHeading() method in AbstractSearchObject. Each AbstractSearchObject subclass could set a different default property, and then the template could just display the result of calling getHeading(). Then it's a single configurable setting instead of an override option.

2.) Maybe you should reverse the order of the heading and ini section settings so it makes more sense to drop the ini section when unsupported. Otherwise you'll have to put in an extra colon separator to override the heading, and that might cause confusion. I also wonder if we should drop the word "hidden" since we're not using hidden filters here, just regular filters. We're filtering the results fetched from the backends, but we don't need to use the hidden filter mechanism since this isn't a normal user-initiated search with faceting/filtering controls. And regarding where filtering is supported, you're correct that it should work for Summon. It should also work for Web, since that's a standard Solr-based search.

Reply:
This all made sense to me.  I've implemented the default heading property as an abstract method so it's easier to check compliance.  Everything else I did as suggested.

I forgot that we removed the client-side filtering settings from EDS.ini recently since they don't work.  So I removed the [ini section] for EDS and EPF, although it's something that's worth investigating along with the larger issue.

I will need help from someone to test the Summon and Web functionality.